patches/0083-lavfi-dnn-Add-OpenVINO-API-2.0-support.patch

```diff
@@ -0,0 +1,1264 @@
++} ov2_errors[] = {
++    { OK,                     0,                  "success"                },
++    { GENERAL_ERROR,          AVERROR_EXTERNAL,   "general error"          },
++    { NOT_IMPLEMENTED,        AVERROR(ENOSYS),    "not implemented"        },
++    { NETWORK_NOT_LOADED,     AVERROR_EXTERNAL,   "network not loaded"     },
++    { PARAMETER_MISMATCH,     AVERROR(EINVAL),    "parameter mismatch"     },
++    { NOT_FOUND,              AVERROR_EXTERNAL,   "not found"              },
++    { OUT_OF_BOUNDS,          AVERROR(EOVERFLOW), "out of bounds"          },
++    { UNEXPECTED,             AVERROR_EXTERNAL,   "unexpected"             },
++    { REQUEST_BUSY,           AVERROR(EBUSY),     "request busy"           },
++    { RESULT_NOT_READY,       AVERROR(EBUSY),     "result not ready"       },
++    { NOT_ALLOCATED,          AVERROR(ENODATA),   "not allocated"          },
++    { INFER_NOT_STARTED,      AVERROR_EXTERNAL,   "infer not started"      },
++    { NETWORK_NOT_READ,       AVERROR_EXTERNAL,   "network not read"       },
++    { INFER_CANCELLED,        AVERROR(ECANCELED), "infer cancelled"        },
++    { INVALID_C_PARAM,        AVERROR(EINVAL),    "invalid C parameter"    },
++    { UNKNOWN_C_ERROR,        AVERROR_UNKNOWN,    "unknown C error"        },
++    { NOT_IMPLEMENT_C_METHOD, AVERROR(ENOSYS),    "not implement C method" },
++    { UNKNOW_EXCEPTION,       AVERROR_UNKNOWN,    "unknown exception"      },
++};
++
++static int ov2_map_error(ov_status_e status, const char **desc)
++{
++    int i;
++    for (i = 0; i < FF_ARRAY_ELEMS(ov2_errors); i++) {
++        if (ov2_errors[i].status == status) {
++            if (desc)
++                *desc = ov2_errors[i].desc;
++            return ov2_errors[i].av_err;
++        }
++    }
++    if (desc)
++        *desc = "unknown error";
++    return AVERROR_UNKNOWN;
++}
++
++static DNNDataType port_datatype_to_dnn_datatype(ov_element_type_e port_datatype)
++{
++    switch (port_datatype)
++    {
++    case F32:
++        return DNN_FLOAT;
++    case U8:
++        return DNN_UINT8;
++    default:
++        av_assert0(!"Not supported yet.");
++        return DNN_FLOAT;
++    }
++}
++
++static int fill_model_input_ov2(OV2Model *ov2_model, OV2RequestItem *request);
++static void infer_completion_callback(void *args);
++
++static int execute_model_ov2(OV2RequestItem *request, Queue *inferenceq)
++{
++    ov_status_e status;
++    LastLevelTaskItem *lltask;
++    int ret = 0;
++    TaskItem *task;
++    OV2Context *ctx;
++    OV2Model *ov2_model;
++
++    if (ff_queue_size(inferenceq) == 0) {
++        ov_infer_request_free(request->infer_request);
++        av_freep(&request);
++        return 0;
++    }
++
++    lltask = ff_queue_peek_front(inferenceq);
++    task = lltask->task;
++    ov2_model = task->model;
++    ctx = &ov2_model->ctx;
++
++    if (task->async) {
++        ret = fill_model_input_ov2(ov2_model, request);
++        if (ret != 0) {
++            goto err;
```

Comment:
Is it needed to free request in the error path ?

Reply:
Oh, I got you, add new free code in the end of this function.